crates/consensus/authority/tests/it/poa_testnet.rs

```diff
@@ -0,0 +1,313 @@
+//! poa-testnet consensus integration test
+use clap::Parser;
+use ethers::{
+    prelude::*,
+    providers::{Http, Provider},
+    signers::LocalWallet,
+};
+use reth::{
+    cli::{
+        components::RethNodeComponents,
+        ext::{NoArgs, NoArgsCliExt, RethNodeCommandConfig},
+    },
+    network::Peers,
+    poa::PoaNodeCommand,
+    runner::CliRunner,
+    tasks::TaskSpawner,
+};
+use reth_primitives::{constants::ETHEREUM_BLOCK_GAS_LIMIT, ChainSpec, BOTANIX_TESTNET};
+use reth_provider::CanonStateSubscriptions;
+use reth_rpc_types::PeerId;
+use secp256k1::SecretKey;
+use std::{
+    collections::HashMap,
+    io::Write,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+    path::{Path, PathBuf},
+    process::Stdio,
+    str::FromStr,
+    sync::Arc,
+    time::Duration,
+};
+use tokio::{
+    io::{self, AsyncBufReadExt},
+    process::Command,
+};
+
+const RPC_PORT_BASE: u16 = 8545;
+const AUTHRPC_PORT_BASE: u16 = 8551;
+const DISCOVERY_PORT_BASE: u16 = 30303;
+const FED_MEMBER1_SECRET_KEY: &'static str =
+    "0a35afe1386497890e1dce7286a5b378b978ede20db900e6ce5b4eb1a0449ad6";
+const FED_MEMBER2_SECRET_KEY: &'static str =
+    "0cc8f5cc52b62b570dc69001f1ab49cd1a7056bf6312fe057f094135f2c9b019";
+const SENDER_SECRET_KEY: &'static str =
+    "52947524bbc14bd90cc86c32b9b7564da2f7f8de343825fed68cd04da4925d29";
+const RECEIVER_ADDRESS: &'static str = "0x613580C865985dA78613Ea7EBCF7a3b8C5445F93";
+
+#[derive(Clone, Debug)]
+struct FederationMemberTestConfig {
+    pub index: u16,
+    pub temp_path: PathBuf,
+    pub secret_key: SecretKey,
+    pub rpc_port: u16,
+    pub authrpc_port: u16,
+    pub discovery_port: u16,
+    pub peers_list: Vec<FederationMemberTestConfig>,
+}
+
+impl FederationMemberTestConfig {
+    pub fn new(index: u16, secret_key: SecretKey) -> Self {
+        Self {
+            index,
+            temp_path: tempfile::TempDir::new().expect("tempdir is okay").into_path(),
+            secret_key,
+            rpc_port: RPC_PORT_BASE + index,
+            authrpc_port: AUTHRPC_PORT_BASE + index,
+            discovery_port: DISCOVERY_PORT_BASE + index,
+            peers_list: vec![],
+        }
+    }
+
+    pub fn insert_piers_list(&mut self, peers: Vec<FederationMemberTestConfig>) {
```

Comment:
Just a little typo here

```suggestion
    pub fn insert_peers_list(&mut self, peers: Vec<FederationMemberTestConfig>) {
```

Reply:
fixed